modules/sdk-coin-eth/src/eth.ts

```diff
@@ -942,6 +1003,239 @@ export class Eth extends BaseCoin {
+
+      if (!userSigningMaterial.backupNShare) {
+        throw new Error('Invalid user key - missing backupNShare');
+      }
+
+      if (!backupSigningMaterial.userNShare) {
+        throw new Error('Invalid backup key - missing userNShare');
+      }
+
+      const MPC = new Ecdsa();
+
+      userKeyCombined = MPC.keyCombine(userSigningMaterial.pShare, [
+        userSigningMaterial.bitgoNShare,
+        userSigningMaterial.backupNShare,
+      ]);
+      backupKeyCombined = MPC.keyCombine(backupSigningMaterial.pShare, [
+        backupSigningMaterial.bitgoNShare,
+        backupSigningMaterial.userNShare,
+      ]);
+
+      const backupKeyPair = new KeyPairLib({ pub: backupKeyCombined.xShare.y });
+
+      backupKeyAddress = backupKeyPair.getAddress();
+    }
+
+    const backupKeyNonce = await this.getAddressNonce(backupKeyAddress);
+
+    // get balance of backupKey to ensure funds are available to pay fees
+    const backupKeyBalance = await this.queryAddressBalance(backupKeyAddress);
+
+    const totalGasNeeded = gasPrice.mul(gasLimit);
+    const weiToGwei = 10 ** 9;
+    if (backupKeyBalance.lt(totalGasNeeded)) {
+      throw new Error(
+        `Backup key address ${backupKeyAddress} has balance ${(backupKeyBalance / weiToGwei).toString()} Gwei.` +
+          `This address must have a balance of at least ${(totalGasNeeded / weiToGwei).toString()}` +
+          ` Gwei to perform recoveries. Try sending some ETH to this address then retry.`
+      );
+    }
+
+    // get balance of wallet and deduct fees to get transaction amount, wallet contract address acts as base address for tss?
+    const txAmount = backupKeyBalance.sub(totalGasNeeded);
+
+    // build recipients object
+    const recipients = [
+      {
+        address: params.recoveryDestination,
+        amount: txAmount.toString(10),
+      },
+    ];
+
+    const txInfo = {
+      recipient: recipients[0],
+      expireTime: this.getDefaultExpireTime(),
+      gasLimit: gasLimit.toString(10),
+    };
+
+    const txParams = {
+      to: params.recoveryDestination, // no contract address, so this field should not be used anyways
+      nonce: backupKeyNonce,
+      value: txAmount,
+      gasPrice: gasPrice,
+      gasLimit: gasLimit,
+      data: Buffer.from('0x'), // no contract call, so no call data?
+      eip1559: params.eip1559,
+      replayProtectionOptions: params.replayProtectionOptions,
+    };
+
+    let tx = Eth.buildTransaction(txParams);
+
+    if (isUnsignedSweep) {
+      return this.formatForOfflineVaultTSS(
+        txInfo,
+        tx,
+        userKey,
+        backupKey,
+        gasPrice,
+        gasLimit,
+        backupKeyNonce,
+        params.eip1559,
+        params.replayProtectionOptions
+      );
+    }
+
+    const signableHex = tx.getMessageToSign(false).toString('hex');
+
+    const signature = this.signRecoveryTSS(userKeyCombined, backupKeyCombined, signableHex);
+
+    // get signed Tx from signature
+    const txData = tx.toJSON();
```

Comment:
can you also extract this into it's own method? this can be used by WP since WP is also doing same thing